src/cmd/linuxkit/pkg_build.go

```diff
@@ -102,9 +130,38 @@ func pkgBuild(args []string) {
 		os.Exit(1)
 	}
 	opts = append(opts, pkglib.WithBuildBuilders(buildersMap))
-	if err := p.Build(opts...); err != nil {
-		fmt.Fprintf(os.Stderr, "%v\n", err)
-		os.Exit(1)
+
+	for _, p := range pkgs {
+		// unless overridden, platforms are specific to a package, so this needs to be inside the for loop
+		if len(plats) == 0 {
```

Comment:
The second time this loops around, won't `plats` now be > 0 and therefore the comment above is no longer true? Effectively if not specified, we'll always build packages to the arches supported in the first package.

Reply:
This is a really good point. I got that with the `fiBase`, but not this one. Fixing.

Reply:
Fixed